stackit/internal/services/argus/instance/resource.go

```diff
@@ -405,6 +623,40 @@ func (r *instanceResource) Create(ctx context.Context, req resource.CreateReques
 		return
 	}
 
+	// Alert Config
+
+	if model.AlertConfig.IsUnknown() || model.AlertConfig.IsNull() {
+		alertConfig = mockAlertConfig
+	}
```

Comment:
We should have some explaination here on why we need to do this

Reply:
Actually, do we need to set the mock on creation? Couldn't it be done just on the update, when the user is trying to delete their alert configs?

Reply:
The advantage of setting our mock on creation instead of relying on the API's mock is that in the first update command after creating the instance only has the changes that the user intends. If we don't set the our mock on creation the user would get the Alert config changes in the plan (trying to change to our mock Alert Config), which may be confusing

Reply:
I think the best middle ground is to have our mock match their current default Alert Config. This way, it will work even if they change their defaults but if they don't, then there will never be "hidden" changes

Reply:
I see, good idea